hack/check-rec-rule-usage.sh

```diff
@@ -1,22 +1,46 @@
 #!/usr/bin/env bash
 set -e
-
 TMP=$(mktemp -d)
 echo "Created temporary directory at $TMP"
 
+# This directory should be a local copy of github.com/openshift/console
+CONSOLE_DIR="${1:-../console}"
+
+if [ ! -d "$CONSOLE_DIR" ]; then
+    echo "Couldn't find a local copy of github.com/openshift/console: directory $CONSOLE_DIR does not exist"
+    exit 1
+fi
+
 findrules="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/find-rules.sh"
 
 "$findrules" >"$TMP/rule-groups"
-# extract recording rules
+# Extract recording rules
 tmp/bin/gojsontoyaml -yamltojson <"$TMP/rule-groups" | jq -s '.. | objects | select(.record) | .record' | tr -d '"' | sort -u >"$TMP/rec-rules"
-# and alerting rules
+
+# Extract alerting rules
 tmp/bin/gojsontoyaml -yamltojson <"$TMP/rule-groups" | jq -s '.. | objects | select(.alert) | .expr' | grep "\w\+{" -o | tr -d "{" | sort -u >"$TMP/used-rules"
-# and telemetry client's metrics
-tmp/bin/gojsontoyaml -yamltojson <manifests/*-config.yaml | jq '.data."metrics.yaml"' | sed "s;\\\\n;\n;g" | grep "^-\s'{__name__=\\\\\"" | sed -e "s/^- '{__name__=\\\\\"//" | sed -e "s/\\\\\"[^}]*}'//" | sort | uniq >>"$TMP/used-rules"
 
-# the first grep outputs all recording rules that are used neither in alerts nor telemetry metrics
+# Extract referenced rules in other rules' expressions
+tmp/bin/gojsontoyaml -yamltojson <"$TMP/rule-groups" | jq -s '.. | objects | select(.expr) | .expr' | sed 's/\\n/\n/g' | sed 's/[^a-zA-Z0-9_]/\n/g' | sort -u | uniq >"$TMP/rule-expr-tokens"
```

Comment:
this part seems brittle to me: looking at the `rule-expr-tokens` file output, it contains way more items than just rules. IMHO the only way we can safely identify whether a recording rule is used by another rule is to parse the PromQL expressions for all rules. This would have to be done in Go using the promql parser (check `pkg/promqlgen/promqlgen.go` for an example).

Reply:
agree, promql parser shall be used to extract referenced rules. I will hold the PR until this modification is done. 
/hold

Reply:
Promql parser is used now to extract referenced rule names.